Test/AbstractWidgetTestCase.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\Test;
+
+use Symfony\Bridge\Twig\Extension\FormExtension;
+use Symfony\Bridge\Twig\Extension\TranslationExtension;
+use Symfony\Bridge\Twig\Form\TwigRenderer;
+use Symfony\Bridge\Twig\Form\TwigRendererEngine;
+use Symfony\Bridge\Twig\Tests\Extension\Fixtures\StubFilesystemLoader;
+use Symfony\Bundle\FrameworkBundle\Tests\Templating\Helper\Fixtures\StubTranslator;
+use Symfony\Component\Form\FormExtensionInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\Form\Test\TypeTestCase;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ *
+ * Base class for tests checking rendering of form widgets.
+ *
+ * You can call $this->renderWidget($formView) to test the form rendering and inspect the outcome html.
+ */
+abstract class AbstractWidgetTestCase extends TypeTestCase
+{
+    /**
+     * @var FormExtensionInterface
+     */
+    private $extension;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $rendererEngine = new TwigRendererEngine(array(
+            'form_div_layout.html.twig',
+        ));
+
+        $csrfProviderClasses = array_filter(array(
+            // symfony <=2.4
```

Comment:
Don't think we need this here, because it's no direct deprecation. Also we have plenty of symfony BC hacks in the code without any comment.